framework/src/main/java/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -3122,7 +3122,25 @@ protected AnnotationMirrorSet getExceptionParameterLowerBoundAnnotations() {
     protected void checkThrownExpression(ThrowTree tree) {
         AnnotatedTypeMirror throwType = atypeFactory.getAnnotatedType(tree.getExpression());
         TypeMirror throwTM = throwType.getUnderlyingType();
+        // Initialize required with the default upper bound annotations
         Set<? extends AnnotationMirror> required = getThrowUpperBoundAnnotations();
+        if (methodTree != null) {
+            List<AnnotatedTypeMirror> allowedExceptions =
+                    atypeFactory.getAnnotatedType(methodTree).getThrownTypes();
+            if (allowedExceptions != null) {
+                for (AnnotatedTypeMirror exception : allowedExceptions) {
+                    Types typesUtil = atypeFactory.getProcessingEnv().getTypeUtils();
+                    if (typesUtil.isSubtype(
+                            exception.getUnderlyingType(), throwType.getUnderlyingType())) {
```

Comment:
What is the logic for this subtype test? Shouldn't it be the other way around?
Why is it only comparing the `TypeMirrors` and not considering annotations?

Reply:
The purpose here is to update `required` variable, which is used to check subtyping relationship between actual throw type and required.

The logic is: we should only update `require` only if the exception throwed is a subtype of one of the decleared throw exception in method signature. For example for the following code, we should update `required` to `@Initailized` instead of use top annotation in the hierarchy. And since actual throw exception has `@UnderInitialization` type. so throw type invalid error should be issued

```
    EISOPIssue363(int dummy1, int dummy2, int dummy3) throws @Initailized MyException {
        // :: error: (throw.type.invalid)
        throw new MyException(this, "UnderInitialization");
    }
```

Reply:
add test case for methodinvocation.